solr/core/src/java/org/apache/solr/handler/admin/api/ZookeeperAPI.java

```diff
@@ -0,0 +1,285 @@
+    final ZookeeperFilesResponse response = instantiateJerseyResponse(ZookeeperFilesResponse.class);
+    final SolrParams params = solrQueryRequest.getParams();
+    Map<String, String> map = new HashMap<>(1);
+    map.put(WT, "raw");
+    map.put(OMIT_HEADER, "true");
+    // solrQueryRequest.setParams(SolrParams.wrapDefaults(new MapSolrParams(map), params));
+    synchronized (this) {
+      if (pagingSupport == null) {
+        pagingSupport = new PagedCollectionSupport();
+        ZkController zkController = coreContainer.getZkController();
+        if (zkController != null) {
+          // get notified when the ZK session expires (so we can clear the cached collections and
+          // rebuild)
+          zkController.addOnReconnectListener(pagingSupport);
+        }
+      }
+    }
+
+    String path = params.get(PATH);
+
+    if (params.get("addr") != null) {
+      throw new SolrException(ErrorCode.BAD_REQUEST, "Illegal parameter \"addr\"");
+    }
+
+    boolean detail = false;
+
+    String dumpS = params.get("dump");
+    boolean dump = dumpS != null && dumpS.equals("true");
+
+    int start = params.getInt("start", 0); // Note start ignored if rows not specified
+    int rows = params.getInt("rows", -1);
+
+    String filterType = params.get("filterType");
+    if (filterType != null) {
+      filterType = filterType.trim().toLowerCase(Locale.ROOT);
+      if (filterType.length() == 0) filterType = null;
+    }
+    FilterType type = (filterType != null) ? FilterType.valueOf(filterType) : FilterType.none;
+
+    String filter = (type != FilterType.none) ? params.get("filter") : null;
+    if (filter != null) {
+      filter = filter.trim();
+      if (filter.length() == 0) filter = null;
+    }
+
+    ZKPrinter printer = new ZKPrinter(coreContainer.getZkController());
+    printer.detail = detail;
+    printer.dump = dump;
+    boolean isGraphView = "graph".equals(params.get("view"));
+    // There is no znode /clusterstate.json (removed in Solr 9), but we do as if there's one and
+    // return collection listing. Need to change services.js if cleaning up here, collection list is
+    // used from Admin UI Cloud - Graph
+    boolean paginateCollections = (isGraphView && "/clusterstate.json".equals(path));
+    printer.page = paginateCollections ? new PageOfCollections(start, rows, type, filter) : null;
+    printer.pagingSupport = pagingSupport;
+
+    try {
+      if (paginateCollections) {
+        // List collections and allow pagination, but no specific znode info like when looking at a
+        // normal ZK path
+        response.zookeeperFiles = printer.printPaginatedCollections();
+      } else {
+        response.zookeeperFiles = printer.print(path);
+      }
+    } finally {
+      printer.close();
+    }
+    return response;
+  }
+
+  @GET
+  @Path("/files/{path}")
+  @Produces({"application/json", "application/xml", BINARY_CONTENT_TYPE_V2})
+  @Operation(
+      summary = "List Zookeeper file.",
```

Comment:
[0] Looks like a copy/paste mistake from the API above?  I'd expect the summary here to be something about fetching a specific single ZK file.